Runtime/Client/LootLockerLifecycleManager.cs

```diff
@@ -0,0 +1,968 @@
+                }
+                finally
+                {
+                    _state = LifecycleManagerState.Ready; // Always reset the state
+                }
+            }
+        }
+
+        /// <summary>
+        /// Service health monitoring coroutine - checks service health and restarts failed services
+        /// </summary>
+        private IEnumerator ServiceHealthMonitor()
+        {
+            const float healthCheckInterval = 30.0f; // Check every 30 seconds
+            
+            while (_serviceHealthMonitoringEnabled && Application.isPlaying)
+            {
+                yield return new WaitForSeconds(healthCheckInterval);
+                
+                if (_state != LifecycleManagerState.Ready)
+                {
+                    continue; // Skip health checks during initialization/reset
+                }
+
+                lock (_serviceLock)
+                {
+                    // Check each service health
+                    var servicesToRestart = new List<Type>();
+                    
+                    foreach (var serviceEntry in _services)
+                    {
+                        var serviceType = serviceEntry.Key;
+                        var service = serviceEntry.Value;
+                        
+                        if (service == null)
+                        {
+                            LootLockerLogger.Log($"Service {serviceType.Name} is null - marking for restart", LootLockerLogger.LogLevel.Warning);
+                            servicesToRestart.Add(serviceType);
+                            continue;
+                        }
+                        
+                        try
+                        {
+                            // Check if service is still initialized
+                            if (!service.IsInitialized)
+                            {
+                                LootLockerLogger.Log($"Service {service.ServiceName} is no longer initialized - attempting restart", LootLockerLogger.LogLevel.Warning);
+                                servicesToRestart.Add(serviceType);
+                            }
+                        }
+                        catch (Exception ex)
+                        {
+                            LootLockerLogger.Log($"Error checking health of service {serviceType.Name}: {ex.Message} - marking for restart", LootLockerLogger.LogLevel.Warning);
+                            servicesToRestart.Add(serviceType);
+                        }
+                    }
+                    
+                    // Restart failed services
+                    foreach (var serviceType in servicesToRestart)
+                    {
+                        _RestartService(serviceType);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Restart a specific service that has failed
+        /// </summary>
+        private void _RestartService(Type serviceType)
```

Comment:
This seems like it should only call Reset and Initialize service in order and nothing else

Reply:
Fixed